### PR TITLE
feat: added favourites section

### LINKED ISSUE #160: add botttom navigation item for favourite events
working on it 👍

### PR DESCRIPTION
**Pull request by simarsingh24**
fixes #160

### CODE REVIEW

**Comment by simarsingh24** [comment]
screen-shot after the changes
image

File: app/src/main/java/org/fossasia/openevent/general/event/EventUtils.kt [hunk 1]
```diff
@@ -72,4 +72,5 @@ object EventUtils {
             return ""
         }
     }
+
```
Comment (nikit19) [inline app/src/main/java/org/fossasia/openevent/general/event/EventUtils.kt]: extra

File: app/src/main/res/menu/navigation.xml [hunk 2]
```diff
@@ -11,9 +11,13 @@
         android:icon="@drawable/ic_search_grey_24dp"
         android:title="@string/search" />
 
+    <item
+        android:id="@+id/navigationFavorite"
+        android:icon="@drawable/ic_baseline_favorite_border_24px"
+        android:title="@string/Profile" />
```
Comment (dreadpool2) [inline app/src/main/res/menu/navigation.xml]: Change this to Favourite :+1:

File: app/src/main/res/layout/fragment_favorite.xml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        </ProgressBar>
+
+        <android.support.v7.widget.RecyclerView
+            android:id="@+id/favoriteEventsRecycler"
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            android:scrollbars="vertical" />
+    </FrameLayout>
+
+
```
Comment (nikit19) [inline app/src/main/res/layout/fragment_favorite.xml]: extra

File: app/src/main/res/menu/navigation.xml (hunk 2 above)
Comment (simarsingh24) [reply app/src/main/res/menu/navigation.xml]: was already doing that

File: app/src/main/res/layout/item_card_favorite_event.xml [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+            <TextView
+                android:id="@+id/favoriteEventDate"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:layout_marginLeft="10dp"
+                android:textSize="17sp"
+                tools:text="Thu, Jun 28" />
+
+
```
Comment (nikit19) [inline app/src/main/res/layout/item_card_favorite_event.xml]: extra

**Comment by dreadpool2** [comment]
UI looks good :+1:

File: app/src/main/res/menu/navigation.xml [hunk 5]
```diff
@@ -11,9 +11,13 @@
...
+
     <item
         android:id="@+id/navigationProfile"
         android:icon="@drawable/ic_baseline_account_circle_24px"
         android:title="@string/Profile" />
-
```
Comment (nikit19) [inline app/src/main/res/menu/navigation.xml]: not neede

**Comment by simarsingh24** [comment]
ss after the change
image

File: app/src/main/java/org/fossasia/openevent/general/favorite/FavoriteEventViewHolder.kt [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+import android.view.View
+import com.squareup.picasso.Picasso
+import kotlinx.android.synthetic.main.item_card_favorite_event.view.*
+import org.fossasia.openevent.general.R
+import org.fossasia.openevent.general.event.Event
+import org.fossasia.openevent.general.event.EventUtils
+import org.fossasia.openevent.general.event.FavoriteFabListener
+import org.fossasia.openevent.general.event.RecyclerViewClickListener
+
+
```
Comment (dreadpool2) [inline app/src/main/java/org/fossasia/openevent/general/favorite/FavoriteEventViewHolder.kt]: extra line

File: app/src/main/java/org/fossasia/openevent/general/favorite/FavoriteEventsRecyclerAdapter.kt [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+import android.support.v7.widget.RecyclerView
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import org.fossasia.openevent.general.R
+import org.fossasia.openevent.general.event.Event
+import org.fossasia.openevent.general.event.FavoriteFabListener
+import org.fossasia.openevent.general.event.RecyclerViewClickListener
+import java.util.*
+
+
```
Comment (dreadpool2) [inline app/src/main/java/org/fossasia/openevent/general/favorite/FavoriteEventsRecyclerAdapter.kt]: extra line

**Comment by simarsingh24** [comment]
@iamareebjamal please review

**Comment by iamareebjamal** [comment]
- Reuse ViewHolder
- Add back button to fragment

**Comment by simarsingh24** [comment]
@iamareebjamal the `item favorite event` UI is different from `item events` UI (mainly the way we display dates) so i cant reuse the view holder and it may happen that later we want to display some more things in favourite event items

**Comment by iamareebjamal** [comment]
What's the difference in dates?

We should try to make UI as consistent and reuse code as much as possible

File: app/src/main/res/values/strings.xml [hunk 8]
```diff
@@ -2,6 +2,7 @@
     <string name="app_name">open-event-android</string>
     <string name="Events">Events</string>
     <string name="Profile">Profile</string>
+    <string name="Likes">Likes</string>
```
Comment (iamareebjamal) [inline app/src/main/res/values/strings.xml]: `likes`

And technically should be favorites

Comment (simarsingh24) [reply app/src/main/res/values/strings.xml]: yes it should be favorites but we are trying to keep it consistent with EventBrite and they call it likes? Should i change this to favourites still?

Comment (iamareebjamal) [reply app/src/main/res/values/strings.xml]: OK, then no

Also does eventbrite show favorites as a separate bottom navigation panel

Comment (simarsingh24) [reply app/src/main/res/values/strings.xml]: yes

Comment (simarsingh24) [reply app/src/main/res/values/strings.xml]: resolved conflicts

**Comment by simarsingh24** [comment]
@iamareebjamal please merge this if no changes are required

**Pull request merged**